src/pip/_internal/resolution/resolvelib/candidates.py

```diff
@@ -179,3 +204,31 @@ def get_install_requirement(self):
         # depend on the base candidate, and we'll get the
         # install requirement from that.
         return None
+
+
+class RequiresPythonCandidate(Candidate):
+    def __init__(self, py_version_info):
+        # type: (Optional[Tuple[int, ...]]) -> None
+        if py_version_info is not None:
+            version_info = normalize_version_info(py_version_info)
+        else:
+            version_info = sys.version_info[:3]
+        self._version = Version(".".join(str(c) for c in version_info))
+
+    @property
+    def name(self):
+        # type: () -> str
+        return "<Python>"  # Avoid conflicting with the PyPI package "Python".
```

Comment:
```suggestion
        return "<Python from Requires-Python>"  # Avoid conflicting with the PyPI package "Python".
```

Slightly more explicit?

Reply:
Applied manually (for linting reasons).